src/wwwroot/controllers/ReportsCtrl.js

```diff
@@ -30,6 +30,9 @@
     $scope.getDailyLimit = function() {
       return $rootScope.accountLimits.daily && $rootScope.accountLimits.daily.limit;
     };
+    $scope.showDailyLimit = function() {
+      return $rootScope.accountLimits.daily;
```

Comment:
Is it possible having a response like this?

```json
{
    "daily": {
        "limit": null,
        "remaining": 0,
        "reset": "2017-09-16T00:00:00Z"
    }
}
```

Because in that case, it will not work.

Also, what do you think about convert the result to boolean to make it clear your intentions?

```js
return !!$rootScope.accountLimits.daily;
```

Reply:
Hi @andresmoschini, about the first comment, is not possible to get a response like that.
This section of the SendRateLimit_To_LimitStatus mapper return null the entire daily property if the limit is null.
```c#
public void Fill(List<SendRateLimit> input, LimitStatus output)
        {
            output.monthly = GetLimitFromSendRateLimit(input.FirstOrDefault(r => r.LimitType == RateLimitType.Monthly));
            output.daily = GetLimitFromSendRateLimit(input.FirstOrDefault(r => r.LimitType == RateLimitType.Daily));
            output.hourly = GetLimitFromSendRateLimit(input.FirstOrDefault(r => r.LimitType == RateLimitType.Hourly));
        }

        private static Limit GetLimitFromSendRateLimit(SendRateLimit sendRateLimit)
        {
            if (sendRateLimit == null || !sendRateLimit.Limit.HasValue)
            {
                return null;
            }
           
            var remaining = (sendRateLimit.Limit.Value - sendRateLimit.CurrentCount);
            return new Limit
            {
                remaining = (remaining < 0) ? 0 : remaining,
                limit = sendRateLimit.Limit.Value,
                reset = sendRateLimit.ResetDate.Value.ToDateTimeOffset()
            };
        }
```

About the second comment, yes! I will do it. Thank you!

Reply:
You are right! also, the [schema](http://api.dopplerrelay.com/schemas/limit-status.json) indicates that _limit_ field is not nullable.

https://github.com/MakingSense/Doppler_Relay/blob/d31706fc2c9cf09ad4dd6b7ae6a335d1791b9b0c/Relay.Application.Api/wwwroot/schemas/limit-status.json#L24-L41

Thanks for the details.

Reply:
Thanks for tip! :)